snuba/query/snql/parser.py

```diff
@@ -807,6 +807,42 @@ def parse(exp: Expression) -> Expression:
     query.transform_expressions(parse)
 
 
+def _mangle_query_aliases(
+    query: Union[CompositeQuery[QueryEntity], LogicalQuery],
+) -> None:
+    """
+    If a query has a subquery, the inner query will get its aliases mangled. This is
+    a problem because the outer query is using the inner aliases, not the inner
+    selected expression values.
+
+    So, we mangle the outer query column names to match the inner query aliases as well.
+    """
+
+    alias_prefix = "_snuba_"
+
+    def mangle_aliases(exp: Expression) -> Expression:
+        alias = exp.alias
+        if alias is not None:
+            return replace(exp, alias=f"{alias_prefix}{alias}")
+
+        return exp
+
+    def mangle_column_value(exp: Expression) -> Expression:
+        if not isinstance(exp, Column):
+            return exp
+
+        return replace(exp, column_name=f"{alias_prefix}{exp.column_name}")
+
+    query.transform_expressions(mangle_aliases)
+
+    # Check if this query has a subquery. If it does, we need to mangle the column name as well
+    # and keep track of what we mangled by updating the mappings in memory.
+    if isinstance(query, CompositeQuery) and not isinstance(
+        query.get_from_clause(), JoinClause
+    ):
```

Comment:
`if isinstance(query.get_from_clause()), Query)`

In other way to avoid the isinstance, you can use the data source visitor 
https://github.com/getsentry/snuba/blob/master/snuba/query/data_source/visitor.py#L11

Reply:
You should be able to simply use 
`if isinstance(query.get_from_clause()), Query)`